go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -289,13 +290,87 @@ func (m *MasterController) handleOverlayPort(node *kapi.Node, annotator kube.Ann
 		}
 	}
 
+	// create a reroute on the ovn_cluster_router
+	err = createLogicalRouterPolicyForNode(node, subnets)
+	if err != nil {
+		return fmt.Errorf("failed to create logical router policy: %v", err)
+	}
+
+	return nil
+}
+
+func createLogicalRouterPolicyForNode(node *kapi.Node, subnets []*net.IPNet) error {
+	// get all the windows subnets as the dest for lr-policy match
+	var dest string
+	var nextHop string
+	var src string
+	for _, hybridSubnet := range config.HybridOverlay.ClusterSubnets {
+		if !utilnet.IsIPv6CIDR(hybridSubnet.CIDR) {
+			if len(dest) > 0 {
+				dest = dest + " || "
+			}
+			dest = dest + fmt.Sprintf("ip4.dst == %s", hybridSubnet.CIDR.String())
+		}
+	}
+	//the src is the nodes subnet and the next hop is the third address (x.x.x.3) in the nodes subnet
+	for _, subnet := range subnets {
+		if !utilnet.IsIPv6CIDR(subnet) {
+			src = fmt.Sprintf("ip4.src == %s", subnet.String())
+			ip := util.GetNodeHybridOverlayIfAddr(subnet)
+			nextHop = ip.IP.String()
+			break
+		}
+	}
+	if len(nextHop) == 0 {
+		return fmt.Errorf("could not find an ipv4 Node subnet for node %s", node.Name)
+	}
+
+	_, stderr, err := util.RunOVNNbctl(
+		"--id=@lr-policy",
+		"create",
+		"logical_router_policy",
+		"priority=1006",
+		"action=reroute",
+		fmt.Sprintf("external-ids:hybrid-overlay=%s", node.Name),
+		fmt.Sprintf("match=\"(%s) && %s\"", dest, src),
```

Comment:
this wont work for hybrid overlay exgw right? You would need to add route policy matching on pods src ip that are in a namespace with the exgw annotation. If we dont care about HO exgw because we now have OVN egress gw then let's remove the code.

Reply:
I see you only removed the .3 route for when HO cluster subnets are involved. Question is do we really want to have the old exgw hybrid overlay code still using .3 route? Should we just remove that code?

Reply:
@dcbw could you weigh in on this?